services/workflows-service/src/customer/customer.service.ts

```diff
@@ -1,20 +1,46 @@
 import { Injectable } from '@nestjs/common';
-import { CustomerRepository } from '@/customer/customer.repository';
 import { Prisma } from '@prisma/client';
-import { TCustomerWithFeatures } from '@/customer/types';
+import dayjs from 'dayjs';
+
+import { UnifiedApiClient } from '@/common/utils/unified-api-client/unified-api-client';
 import { ApiKeyService } from '@/customer/api-key/api-key.service';
 import { generateHashedKey } from '@/customer/api-key/utils';
-import { UnifiedApiClient } from '@/common/utils/unified-api-client/unified-api-client';
-import { PrismaService } from '@/prisma/prisma.service';
+import { CustomerRepository } from '@/customer/customer.repository';
+import { TCustomerWithFeatures } from '@/customer/types';
 import { env } from '@/env';
+import { MerchantMonitoringClient } from '@/merchant-monitoring/merchant-monitoring.client';
+import { PrismaService } from '@/prisma/prisma.service';
+import { TDemoAccessDetails } from './schemas/zod-schemas';
 
 @Injectable()
 export class CustomerService {
   constructor(
     protected readonly repository: CustomerRepository,
     protected readonly apiKeyService: ApiKeyService,
     private readonly prisma: PrismaService,
+    private readonly merchantMonitoringClient: MerchantMonitoringClient,
   ) {}
+
+  async getDemoAccessDetails(customer: TCustomerWithFeatures): Promise<TDemoAccessDetails | null> {
+    const { id: customerId, config, createdAt } = customer;
+
+    if (!config || !config.isDemoAccount || !createdAt) {
+      return null;
+    }
+
+    const businessReportsCount = await this.merchantMonitoringClient.count({
+      customerId,
+    });
+
+    return {
+      totalReports: businessReportsCount,
```

Comment:
Do we have a mechanism in the front-end the ensure this property gets updated when reports are added?

Reply:
And just a quick comment: Ideally, this shouldn’t be under customers and should have its own endpoint, like `GET /api/business-reports/count`. But since it’s just for a demo, it's not a big deal